dashboard/app/controllers/home_controller.rb

```diff
@@ -42,7 +42,27 @@ def index
       @recent_courses = recent_courses.slice(0, 2)
 
       if current_user.teacher?
-        @sections = []
+        base_url = CDO.code_org_url('/teacher-dashboard#/sections/')
+        @sections = current_user.sections.map do |section|
+          if section.script_id
+            course_name = Script.find_by_id(section.script_id)[:name]
```

Comment:
Is there a reason this needs to hit the DB rather than the cache? E.g., any reason not to do the following?
```
course_name = Script.get_from_cache(section.script_id)[:name]
```